### DISCUSSION TITLE
How to stream h264 video (no sound) to web browser with low latency?

### DISCUSSION THREAD

**Discussion by xdanik** [bluenviron/mediamtx#338] [Questions]
Hello everybody,

I am looking for a way to stream h264 (1080p@15fps, no audio, keyframe every second) with low latency (Ideally less than a second) to a arbitrary number of web clients.

The most commonly used technologies for streaming video to the browser are HLS and DASH, but both have fairly high latency.

I am currently using Shinobi CCTV just as a streaming proxy because of its "Poseidon" websocket streaming capability. But it seems little overkill to run full fat CCTV solution just for a simple video proxy.
Also I am having issues with Posiedon streams as they are sometime not playing P/B-frames witch results in video made just from  keyframes - slideshow 😉.

Any tips for a software that can be used for that? (Ideally free and opensource with Linux x64 support)

**Comment by aler9** [comment]
> Also I am having issues with Posiedon streams as they are sometime not playing P/B-frames witch results in video made just from keyframes - slideshow wink.

Exactly, this is the problem of websocket-based web streaming - it's fast, but B frames create a mess.

Short answer: re-encode in h264 baseline and use [Broadway](https://github.com/mbebenita/Broadway) - you obtain latencies below 200ms, but you also need some computational power to perform the re-encoding.

Otherwise, you can wait until i've added HLS and low-latency HLS to rtsp-simple-server (that's what i'm working on), that should allow low latencies even with B-frames, but it will take a while.

**Reply by aler9** [reply]
HLS support is coming - reading with HLS is supported in main, and proxying HLS is coming soon.

[a comment by an automated account is not shown]
